src/lib/expansion/expansion-panel.ts

```diff
@@ -78,6 +80,18 @@ export class MatExpansionPanel extends CdkAccordionItem
   }
   private _hideToggle = false;
 
+  /** An event emitted before the body's expansion animation happens. */
+  @Output() beforeExpanded = new EventEmitter<null>();
```

Comment:
I'm not sure whether emitting `null` is particularly useful. We should either switch it to `void` or emit the `AnimationEvent`.

Reply:
I definitely meant `void` and then typed `null`